tests/generic-java-signatures/derivedNames.scala fails on macOS

## Compiler version

3.0.0

## Minimized example

```sh
sbt testCompilation derivedNames
```

## Output
```
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at dotty.tools.vulpix.ChildJVMMain.runMain(ChildJVMMain.java:40)
        at dotty.tools.vulpix.ChildJVMMain.main(ChildJVMMain.java:47)
Caused by: java.lang.AssertionError: assertion failed: Test$Foo$A<Test$Foo<T1>$B$> != Test$Foo.Test$Foo$A<Test.Test$Foo<T1>.B$>
        at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
        at Test$.main(derivedNames.scala:12)
        at Test.main(derivedNames.scala)
        ... 6 more
```

## Expectation
No test failed.

--- a/tests/generic-java-signatures/derivedNames.scala
+++ b/tests/generic-java-signatures/derivedNames.scala
@@ -4,12 +4,10 @@ object Test {
   def main(args: Array[String]): Unit = {
     val objectB = classOf[Foo[Any]].getClasses
     val returnType = objectB(1).getDeclaredMethod("m").getGenericReturnType.asInstanceOf[ParameterizedType]
-    val out1 = "Test$Foo.Test$Foo$A<Test.Test$Foo<T1>.B$>" // Windows and OSX
-    val out2 = "Test$Foo$A<Test$Foo<T1>$B$>"               // Linux and sometimes Windows
+    val out1 = "Test$Foo.Test$Foo$A<Test.Test$Foo<T1>.B$>" // Windows
+    val out2 = "Test$Foo$A<Test$Foo<T1>$B$>"               // Linux, OSX and sometimes Windows
     if (scala.util.Properties.isWin)
       assert(returnType.toString == out1 || returnType.toString == out2)
-    else if (scala.util.Properties.isMac)
-      assert(returnType.toString == out1, s"$returnType != $out1")
     else
       assert(returnType.toString == out2)
   }

